### PR TITLE
fix: added unit test and updated comments

### PR DESCRIPTION
**Pull request by i-chvets**
Summary of changes:
- Added comments to S3 service code to clarify the usage.
- Added unit test to ensure that S3 URL endpoint is generated correctly in S3 service and in charm code.

### CODE REVIEW

File: charms/mlflow-server/src/charm.py [hunk 1]
```diff
@@ -331,6 +331,10 @@ def _define_secrets(self, obj_storage, mysql):
             {"name": f"{self.charm_name}-db-secret", "data": _db_secret_dict(mysql=mysql)},
         ]
 
+    def _gen_obj_storage_endpoint_url(self, obj_storage):
+        """Generate object storage endpoint URL."""
+        return f"http://{obj_storage['service']}.{obj_storage['namespace']}:{obj_storage['port']}"
```
Comment (DnPlas) [inline charms/mlflow-server/src/charm.py:336]: Should we add a check to ensure  `obj_storage`  actually has those keys you are calling?

File: charms/mlflow-server/src/charm.py [hunk 2]
```diff
@@ -331,6 +331,10 @@ def _define_secrets(self, obj_storage, mysql):
             {"name": f"{self.charm_name}-db-secret", "data": _db_secret_dict(mysql=mysql)},
         ]
 
+    def _gen_obj_storage_endpoint_url(self, obj_storage):
```
Comment (DnPlas) [inline charms/mlflow-server/src/charm.py:334]:
```suggestion
   @property
    def _gen_obj_storage_endpoint_url(self, obj_storage):
```

File: charms/mlflow-server/tests/unit/test_operator.py [hunk 3]
```diff
@@ -464,3 +465,18 @@ def test_install_without_nodeport_and_lb_services(harness, mocker):
     assert harness.charm.model.unit.status == ActiveStatus()
 
     assert pod_spec[0]["kubernetesResources"]["services"] == []
+
+
+def test_storage_endpoint_generation(harness, sample_object_storage):
+    """Test S3 storage URL for correct generation in S3 service and in charm code."""
+    harness.begin()
+    s3_wrapper = S3BucketWrapper(
```
Comment (DnPlas) [inline charms/mlflow-server/tests/unit/test_operator.py:473]: Is there a way to mock this out?

**Review by DnPlas — COMMENTED** [review]
Thanks @i-chvets, left some comments.

**Review by DnPlas — COMMENTED** [review]
Thanks @i-chvets, left some comments.

File: charms/mlflow-server/src/charm.py (hunk 1 above)
Comment (i-chvets) [reply charms/mlflow-server/src/charm.py:336]: Older code never had those checks. But I guess it is good idea. Will return empty string then.

File: charms/mlflow-server/src/charm.py (hunk 2 above)
Comment (i-chvets) [reply charms/mlflow-server/src/charm.py:335]: This is not really a property. It is a generating function. So, I would not say this should be here.

File: charms/mlflow-server/src/charm.py (hunk 1 above)
Comment (i-chvets) [reply charms/mlflow-server/src/charm.py:337]: Done.

**Review by DnPlas — APPROVED** [review]

**Pull request merged**